Plugins/BTCPayServer.RockstarDev.Plugins.TransactionCounter/Services/TxCounterService.cs

```diff
@@ -58,46 +65,65 @@ private async Task<int> FetchAndCacheTransactionCount(CounterPluginSettings mode
             Status = new[] { InvoiceStatus.Processing.ToString(), InvoiceStatus.Settled.ToString() },
             StoreId = includedStoreIds.Length > 0 ? includedStoreIds : allStoreIds
         };
-        var transactionCount = await _invoiceRepository.GetInvoiceCount(query);
-        var total = transactionCount + CalculateExtraTransactionCount(model);
+        var transactions = await _invoiceRepository.GetInvoices(query);
```

Comment:
This algorithm is highly inefficient because it retrieves all invoices from the database, storing them in memory before performing calculations.

Instead, we should write a query that fetches the results directly from the database, eliminating any unnecessary in-memory processing.